pkg/detectors/azure_entra/refreshtoken/refreshtoken.go

```diff
@@ -0,0 +1,312 @@
+package refreshtoken
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/golang-jwt/jwt/v4"
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	logContext "github.com/trufflesecurity/trufflehog/v3/pkg/context"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors/azure_entra"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+	detectors.DefaultMultiPartCredentialProvider
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ interface {
+	detectors.Detector
+	detectors.MaxSecretSizeProvider
+	detectors.StartOffsetProvider
+} = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	refreshTokenPat = regexp.MustCompile(`\b0\.A[\w-]{50,}\.Ag[\w-]{250,}`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"0.A"}
+}
+
+func (s Scanner) Type() detectorspb.DetectorType {
+	return detectorspb.DetectorType_AzureRefreshToken
+}
+
+func (s Scanner) Description() string {
+	return "Azure Entra ID refresh tokens provide long-lasting access to an account."
+}
+
+func (Scanner) MaxSecretSize() int64 { return 2048 }
+
+func (Scanner) StartOffset() int64 { return 4096 }
```

Comment:
@ahrav In this case the token can be quite long (1000+) and the other fields can be either before or after it. What's the correct way to look +/- ~1024 on either side?